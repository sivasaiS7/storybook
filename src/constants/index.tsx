export enum AlertIconEnum {
  ERROR = "error",
  SUCCESS = "success",
  LOADING = "loading",
  INFO = "info",
}

export enum NotificationIconEnum {
  DEFAULT = "default",
  FILL = "fill",
  DOT = "dot",
  DOT_FILL = "dotFill",
}

export enum IconSizeEnum {
  SMALL = "small",
  MEDIUM = "medium",
  LARGE = "large",
}
