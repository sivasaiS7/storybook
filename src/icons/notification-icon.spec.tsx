import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";

import { NotificationIconEnum, IconSizeEnum } from "../constants";
import Theme from "../styles/theme";
import NotificationIcon from "./notification-icon";

describe("Notification Icon component", () => {
  it("applies default styles", () => {
    render(<NotificationIcon />);
    expect(screen.getByAltText("notification-bell-icon")).toMatchSnapshot();
  });

  it("applies DOT variant styles", () => {
    render(<NotificationIcon icon={NotificationIconEnum.DOT} />);
    expect(screen.getByAltText("notification-bell-icon")).toMatchSnapshot();
  });

  it("applies FILL variant styles", () => {
    render(<NotificationIcon icon={NotificationIconEnum.FILL} />);
    expect(screen.getByAltText("notification-bell-icon")).toMatchSnapshot();
  });

  it("applies DOT_FILL variant styles", () => {
    render(<NotificationIcon icon={NotificationIconEnum.DOT_FILL} />);
    expect(screen.getByAltText("notification-bell-icon")).toMatchSnapshot();
  });

  it("applies SMALL size styles", () => {
    render(<NotificationIcon size={IconSizeEnum.SMALL} />);
    expect(screen.getByAltText("notification-bell-icon")).toMatchSnapshot();
  });

  it("applies MEDIUM size styles", () => {
    render(<NotificationIcon size={IconSizeEnum.MEDIUM} />);
    expect(screen.getByAltText("notification-bell-icon")).toMatchSnapshot();
  });

  it("applies LARGE size styles", () => {
    render(<NotificationIcon size={IconSizeEnum.LARGE} />);
    expect(screen.getByAltText("notification-bell-icon")).toMatchSnapshot();
  });

  it("applies filter styles", () => {
    render(<NotificationIcon filter={Theme.filterColors.singlifeRed} />);
    expect(screen.getByAltText("notification-bell-icon")).toMatchSnapshot();
  });
});
