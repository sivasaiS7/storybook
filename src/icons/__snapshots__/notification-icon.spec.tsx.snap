// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Icon component applies DOT variant styles 1`] = `
.emotion-0 {
  width: 1rem;
}

<img
  alt="notification-bell-icon"
  class="notification-icon-image emotion-0"
  data-testid="dot-notification-bell-icon"
  src="notification-dot.svg"
/>
`;

exports[`Notification Icon component applies DOT_FILL variant styles 1`] = `
.emotion-0 {
  width: 1rem;
}

<img
  alt="notification-bell-icon"
  class="notification-icon-image emotion-0"
  data-testid="dotFill-notification-bell-icon"
  src="notification-dot-fill.svg"
/>
`;

exports[`Notification Icon component applies FILL variant styles 1`] = `
.emotion-0 {
  width: 1rem;
}

<img
  alt="notification-bell-icon"
  class="notification-icon-image emotion-0"
  data-testid="fill-notification-bell-icon"
  src="notification-fill.svg"
/>
`;

exports[`Notification Icon component applies LARGE size styles 1`] = `
.emotion-0 {
  width: 2rem;
}

<img
  alt="notification-bell-icon"
  class="notification-icon-image emotion-0"
  data-testid="default-notification-bell-icon"
  src="notification.svg"
/>
`;

exports[`Notification Icon component applies MEDIUM size styles 1`] = `
.emotion-0 {
  width: 1.5rem;
}

<img
  alt="notification-bell-icon"
  class="notification-icon-image emotion-0"
  data-testid="default-notification-bell-icon"
  src="notification.svg"
/>
`;

exports[`Notification Icon component applies SMALL size styles 1`] = `
.emotion-0 {
  width: 1rem;
}

<img
  alt="notification-bell-icon"
  class="notification-icon-image emotion-0"
  data-testid="default-notification-bell-icon"
  src="notification.svg"
/>
`;

exports[`Notification Icon component applies default styles 1`] = `
.emotion-0 {
  width: 1rem;
}

<img
  alt="notification-bell-icon"
  class="notification-icon-image emotion-0"
  data-testid="default-notification-bell-icon"
  src="notification.svg"
/>
`;

exports[`Notification Icon component applies filter styles 1`] = `
.emotion-0 {
  width: 1rem;
  -webkit-filter: invert(18%) sepia(82%) saturate(6823%) hue-rotate(353deg) brightness(100%) contrast(115%);
  filter: invert(18%) sepia(82%) saturate(6823%) hue-rotate(353deg) brightness(100%) contrast(115%);
}

<img
  alt="notification-bell-icon"
  class="notification-icon-image emotion-0"
  data-testid="default-notification-bell-icon"
  src="notification.svg"
/>
`;
